README.md

```diff
@@ -0,0 +1,157 @@
+`credentialsFile`, `tokensFile`, `sheetsURL` and along with a configurable pollingPeriod(Optional).
+
+The `Configure` method is called to parse the configurations. After which, the `Open` method is called to start the connection from the provided position.
+
+
+### Record Fetching
+
+Upon successful connection, an api hit will fetch all the records present in the sheet. If no more data is available,
+the iterator will programmatically pause till the configurable pollingPeriod duration(which has a default value of "6s") and on completion of this duration,
+the api is hit again to fetch the newly added records/rows.
+
+If there are single/multiple empty rows in between the two records, it will fetch only the last record before the first empty row,
+and will hold that position until a new row/record has been added.
+
+
+### Position Handling
+
+The Google Sheets connector stores the last row of the fetched sheet data as position.
+If in case, there are empty row(s), the Sheets connector will fetch till the last non-empty row and that last row will be stored as in position.
+
+
+### Configuration
+
+The config passed to `Configure` in source connector can contain the following fields.
+
+| name                       | description                                                                                                                    | required | example                                                            |
+|----------------------------|--------------------------------------------------------------------------------------------------------------------------------|---------|--------------------------------------------------------------------|
+| `credentialsFile`          | Path to credentials file which can be downloaded from Google Cloud Platform(in .json format) to authorise the user.            | yes     | "path://to/credential/file"                                        |
+| `tokensFile`               | Path to file in .json format which includes the `access_token`, `token_type`, `refresh_token` and `expiry`.                    | yes     | "path://to/token/file"                                             |
+| `sheetsURL`                | URL of the google spreadsheet(copy the entire url from the address bar).                                                       | yes     | "https://docs.google.com/spreadsheets/d/dummy_spreadsheet_id/edit#gid=0" |
+| `DateTimeRenderOption`     | Format of the Date/time related values. Valid values: SERIAL_NUMBER, FORMATTED_STRING                                          | no      | "FORMATTED_STRING"                                                 |
+| `ValueRenderOption`        | Format of the dynamic/reference data. Valid values: FORMATTED_VALUE, UNFORMATTED_VALUE, FORMULA                                | no      | "FORMATTED_VALUE"                                                  |
```

Comment:
Not camel case. ^ I'd usually say it's a nitpick, but given that it's part of the configs, i.e. public, it's better to have it good right away.